rmw_zenoh_cpp/src/detail/zenoh_utils.hpp

```diff
@@ -41,26 +42,24 @@ class ZenohSession final
 };
 
 ///=============================================================================
-void
-create_map_and_set_sequence_num(
-  z_owned_bytes_t * out_bytes, int64_t sequence_number,
-  uint8_t gid[RMW_GID_STORAGE_SIZE]);
+zenoh::Bytes create_map_and_set_sequence_num(
+  int64_t sequence_number, std::vector<uint8_t> gid);
 
 ///=============================================================================
 // A class to store the replies to service requests.
 class ZenohReply final
 {
 public:
-  ZenohReply(const z_loaned_reply_t * reply, std::chrono::nanoseconds::rep received_timestamp);
+  ZenohReply(const zenoh::Reply & reply, std::chrono::nanoseconds::rep received_timestamp);
 
   ~ZenohReply();
 
-  std::optional<const z_loaned_sample_t *> get_sample() const;
+  const zenoh::Reply & get_sample() const;
 
   std::chrono::nanoseconds::rep get_received_timestamp() const;
 
 private:
-  z_owned_reply_t reply_;
+  std::optional<zenoh::Reply> reply_;
```

Comment:
Does this need to be an `optional`?  I kind of think that we always have to have one of these (given the constructor), but it may be the case that we can't construct an empty `zenoh::Reply` object.